lib/block-supports/layout.php

```diff
@@ -89,14 +89,17 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 	return $content . '<style>' . $style . '</style>';
 }
 
-// Register the block support.
-WP_Block_Supports::get_instance()->register(
-	'layout',
-	array(
-		'register_attribute' => 'gutenberg_register_layout_support',
-	)
-);
-add_filter( 'render_block', 'gutenberg_render_layout_support_flag', 10, 2 );
+// This can be removed when plugin support requires WordPress 5.8.0+.
+if ( ! function_exists( 'wp_render_layout_support_flag' ) ) {
+	// Register the block support.
+	WP_Block_Supports::get_instance()->register(
+		'layout',
```

Comment:
Ideally this "key" should be unique and override the existing block support, maybe something to improve in the registry in Core later.

Reply:
Unique key makes sense.

The main problem here is the filters, so I used a single conditional wrapper for both.

Reply:
I wonder if we should use conditionals or actually override the logic in core in case it changes in the plugin. I think we can probably start with ignoring these changes first though.

Reply:
Block support from the plugin should override one from the core. We can probably use higher priority for filters when we need to override the logic in the core.

https://github.com/WordPress/wordpress-develop/blob/81e1c2d58b8134ddf633a64e4dd0bcb5f94856e2/src/wp-includes/class-wp-block-supports.php#L79